### ISSUE TITLE
Missing error when an attribute is a variant generic

### ISSUE THREAD

**Issue by parched** [python/mypy#13185]
The following produces no errors but I expected it to be treated the same as if `B.a` was a get/set property (which causes a variance error)

```python
class A:
    pass

class A2(A):
    something: str = "hello"
    
T_co = TypeVar('T_co', covariant=True)

class B(Generic[T_co]):
    def __init__(self, a: T_co) -> None:
        self.a: T_co = a  # EXPECTED: typing error here like "public attributes must have invariant type"
        
def set_a(b: B[A]) -> None:
    b.a = A()
    

b2: B[A2] = B(A2())

set_a(b2)

print(b2.a.something) # ACTUAL: runtime error here
```

Python 3.10, mypy 0.961 with `--strict`

**Comment by erictraut** [comment]
The problem here is that you have defined `T_co` as covariant, but given how you are using it in `B`, it should be declared as invariant.

**Comment by parched** [comment]
Or is variance something that the user needs to check is correct and it's use should be flagged like `Any` as potentially "unsafe"?

**Comment by erictraut** [comment]
Mypy validates the variance for protocol classes but not for other standard (nominal) classes.
